example/src/main/java/SearchPostsStream.java

```diff
@@ -44,8 +44,9 @@ private void iterateAllPosts(Result result) {
             return;
         }
         Post lastPost = result.getPosts().get(result.getPosts().size() - 1);
-        query.setStartTime(lastPost.getPublished());
-        Result newResult = client.makeRequest(query);
+        Date lastPublishedAt = lastPost.getPublishedAt();
+        q = q.toBuilder().startTime(lastPublishedAt).build();
+        Result newResult = client.makeRequest(q);
```

Comment:
@roback It's actually not the best idea to allow passing `QueryBuilder` into the client.

The reason for the `Query` object existence is that it is immutable. While `QueryBuilder` is mutable and we should try not pass mutable data to the client (as mutability could lead to unpredictable side effects).

I'd prefer to leave it as is. If you do insist - I'll change the implementation.